### ISSUE TITLE
`withTaskCancellationHandler(handler:operation:)` is deprecated

### ISSUE THREAD

**Issue by Krakert** [rickclephas/KMP-NativeCoroutines#107]
In a project Im using `1.0.0-ALPHA-5` and it complains that  `withTaskCancellationHandler(handler:operation:)` is deprecated.
For `AsynFunction` and `AsynSequence`.
```
'withTaskCancellationHandler(handler:operation:)' is deprecated: 
renamed to 'withTaskCancellationHandler(operation:onCancel:)'

Use 'withTaskCancellationHandler(operation:onCancel:)' instead
```
Is this on purpose or is there something wrong with the project?

**Comment by rickclephas** [comment]
Hi! Thanks for the report. This wasn't really on purpose.
Somewhere along the way Apple changed the signature of this function and recently deprecated the old one.
The next release will use the non deprecated version 😄.

**Comment by Krakert** [comment]
Thanks for the quick fix! It sort of stood out because the compiler gave me those two errors.
Because I use Koin in my project I'm on Kotlin `1.8.1`, do I also get the update then?

**Comment by rickclephas** [comment]
The next release will probably come with Kotlin 1.8.21. However since there haven't been any API changes in KMP-NativeCoroutines you could just update the Swift implementation (while keeping an older version for your Kotlin code).

**Comment by rickclephas** [comment]
FYI the fix has been published in [v1.0.0-ALPHA-8](https://github.com/rickclephas/KMP-NativeCoroutines/releases/tag/v1.0.0-ALPHA-8).
